pubs/apis.py

```diff
@@ -25,3 +27,33 @@ def isbn2bibtex(isbn):
     citation = soup.find("textarea").text
 
     return citation
+
+
+def arxiv2bibtex(arxiv_id):
+    """Return a bibtex string of metadata from an arXiv ID"""
+
+    url = 'https://export.arxiv.org/api/query?id_list=' + arxiv_id
+    r = requests.get(url)
+    feed = feedparser.parse(r.text)
+    entry = feed.entries[0]
+
+    if 'title' not in entry:
+        ui = get_ui()
+        ui.error('malformed arXiv ID: {}'.format(arxiv_id))
+        bibtex = None
+    elif 'arxiv_doi' in entry:
+        bibtex = doi2bibtex(entry['arxiv_doi'])
+    else:
+        # Create a bibentry from the metadata.
```

Comment:
Here you can use `bwriter` from the `bibtexparser` package (see https://github.com/pubs/pubs/blob/master/pubs/endecoder.py#L85); this would more likely yield a well-formatted bibtex (be more robust to escaping characters, etc.)

Reply:
Good idea.  I've now constructed a `BibDatabase` and then returned the result from running `bibtexparser.dumps` on it.